File: app/src/main/java/com/example/quoter/data/QuoteRepository.kt
package com.example.quoter.data

import android.content.Context
import java.io.File // Ensure File import is present
import kotlinx.coroutines.flow.Flow

class QuoteRepository(context: Context) {
    private val db = AppDatabase.getDatabase(context)
    private val generatedImageDao = db.generatedImageDao() // Get instance of the new DAO

    // --- Generated Image Methods ---
    fun getAllGeneratedImages(): Flow<List<GeneratedImage>> = generatedImageDao.getAllImages()

    fun getSelectedImagesForRotation(): Flow<List<GeneratedImage>> = generatedImageDao.getSelectedImagesForRotation()

    suspend fun insertGeneratedImage(image: GeneratedImage): Long = generatedImageDao.insertImage(image)

    suspend fun deleteGeneratedImageById(id: Int) = generatedImageDao.deleteImageById(id)

    suspend fun updateGeneratedImageSelection(id: Int, isSelected: Boolean) = generatedImageDao.updateSelectionStatus(id, isSelected)

    suspend fun getRandomSelectedImage(): GeneratedImage? = generatedImageDao.getRandomSelectedImage()

    // Add function to get the first selected image
    suspend fun getFirstSelectedImage(): GeneratedImage? = generatedImageDao.getFirstSelectedImage()

    // Add function to get all selected images ordered by ID
    suspend fun getSelectedImagesOrderedById(): List<GeneratedImage> = generatedImageDao.getSelectedImagesOrderedById()

    suspend fun deleteGeneratedImageFile(context: Context, image: GeneratedImage) {
        // Delete the actual file from internal storage using filePath
        try {
            val file = File(image.filePath) // Use filePath
            if (file.exists()) {
                if (file.delete()) {
                    println("Deleted image file: ${image.filePath}")
                } else {
                    println("Failed to delete image file: ${image.filePath}")
                }
            } else {
                println("Image file not found, skipping deletion: ${image.filePath}")
            }
        } catch (e: Exception) {
            println("Error deleting image file ${image.filePath}: ${e.message}")
            e.printStackTrace()
        }

        // Always delete the database record
        generatedImageDao.deleteImage(image)
    }

    // Add function to delete multiple images (files and DB records)
    suspend fun deleteGeneratedImages(context: Context, images: List<GeneratedImage>) {
        // Delete files first
        images.forEach { image ->
            try {
                val file = File(image.filePath)
                if (file.exists()) {
                    if (file.delete()) {
                        println("Deleted image file: ${image.filePath}")
                    } else {
                        println("Failed to delete image file: ${image.filePath}")
                    }
                } else {
                    println("Image file not found, skipping deletion: ${image.filePath}")
                }
            } catch (e: Exception) {
                println("Error deleting image file ${image.filePath}: ${e.message}")
                e.printStackTrace()
            }
        }
        // Then delete all database records at once
        generatedImageDao.deleteImages(images)
    }
}

File: swift_files/QuoteRepository.swift
swift
import Foundation
import CoreData
import SwiftUI

class QuoteRepository {
    private let context: NSManagedObjectContext
    private let generatedImageDao: GeneratedImageDao

    init(context: NSManagedObjectContext) {
        self.context = context
        self.generatedImageDao = GeneratedImageDao(context: context)
    }

    // --- Generated Image Methods ---
    func getAllGeneratedImages() -> AnyPublisher<[GeneratedImage], Never> {
        return generatedImageDao.getAllImages()
    }

    func getSelectedImagesForRotation() -> AnyPublisher<[GeneratedImage], Never> {
        return generatedImageDao.getSelectedImagesForRotation()
    }

    func insertGeneratedImage(image: GeneratedImage) throws {
         try generatedImageDao.insertImage(image: image)
    }

    func deleteGeneratedImageById(id: UUID) throws {
         try generatedImageDao.deleteImageById(id: id)
    }

    func updateGeneratedImageSelection(id: UUID, isSelected: Bool) throws {
        try generatedImageDao.updateSelectionStatus(id: id, isSelected: isSelected)
    }

    func getRandomSelectedImage() throws -> GeneratedImage? {
        return try generatedImageDao.getRandomSelectedImage()
    }
    
    func getFirstSelectedImage() throws -> GeneratedImage? {
        return try generatedImageDao.getFirstSelectedImage()
    }

    func getSelectedImagesOrderedById() throws -> [GeneratedImage] {
        return try generatedImageDao.getSelectedImagesOrderedById()
    }

    func deleteGeneratedImageFile(image: GeneratedImage) throws {
        // Delete the actual file from internal storage using filePath
        guard let fileURL = image.filePath else {
            print("Error: filePath is nil for image with ID: \(image.id)")
            throw NSError(domain: "com.example.app", code: -1, userInfo: [NSLocalizedDescriptionKey: "filePath is nil"])
        }
        
        do {
            if FileManager.default.fileExists(atPath: fileURL.path) {
                try FileManager.default.removeItem(at: fileURL)
                print("Deleted image file: \(fileURL)")
            } else {
                print("Image file not found, skipping deletion: \(fileURL)")
            }
        } catch {
            print("Error deleting image file \(fileURL): \(error.localizedDescription)")
            throw error
        }
       try generatedImageDao.deleteImage(image: image)
    }

    // Add function to delete multiple images (files and DB records)
    func deleteGeneratedImages(images: [GeneratedImage]) throws {
        // Delete files first
        for image in images {
           try deleteGeneratedImageFile(image: image)
        }
        
        // Then delete all database records at once
        try generatedImageDao.deleteImages(images: images)
    }
}